code/game/machinery/pipe/construction.dm

```diff
@@ -167,23 +167,29 @@ Buildable meters
 			to_chat(user, span_warning("Something is hogging the tile!"))
 			return TRUE
 
-		if(pipe_count == 1 && istype(machine, /obj/machinery/atmospherics/pipe/smart) && ispath(pipe_type, /obj/machinery/atmospherics/pipe/smart) && lowertext(machine.pipe_color) != lowertext(pipe_color) && machine.connection_num < 3)
-			var/direction = machine.dir
-			if((direction & EAST|WEST || direction & SOUTH|NORTH) && !ISDIAGONALDIR(direction))
-				pipe_type = /obj/machinery/atmospherics/pipe/bridge_pipe
-				if(EWCOMPONENT(direction))
-					dir = NORTH
-				if(NSCOMPONENT(direction))
-					dir = EAST
-				continue
-
-		if(flags & PIPING_BRIDGE && !(machine.pipe_flags & PIPING_BRIDGE) && check_ninety_degree_dir(machine)) //continue if we are placing a bridge pipe over a normal pipe only (prevent duplicates)
+		// skip checks if we don't overlap layers, either by being on the same layer or by something being on all layers
+		if((machine.piping_layer != piping_layer) && !((machine.pipe_flags | flags) & PIPING_ALL_LAYER))
 			continue
 
-		if((machine.piping_layer != piping_layer) && !((machine.pipe_flags | flags) & PIPING_ALL_LAYER)) //don't continue if either pipe goes across all layers
+		// If we're a smart pipe, and we're looking to share space with a potentially perpendicular kind of pipe, see if we can smartly go over it as a bridge
+		if(pipe_count == 1 && machine.connection_num < 3 && ispath(pipe_type, /obj/machinery/atmospherics/pipe/smart))
```

Comment:
This connection_num check does nothing. I'm going to merge https://github.com/tgstation/tgstation/pull/60804 because it makes the variable actually change ever, which I assume causes issues here.

This is partially because I feel bad just closing the guy's pr cause you came along and opened a separate fix, and partially because I think it's valuable

Reply:
The `connection_num` check is redundant, as anything with more than two connections is going to be caught by the immediate `!ISDIAGONAL` check.

`git grep connection_num` indicates that this variable is used literally nowhere else, so the lack of variable maintenance doesn't cause issues here or anywhere else, and the variable should be safe be purge. `dir` can be updated at a point where `bitfield` is identical to `connections` and `bits` is identical to `connection_num`, so there's not much point in building two copies of everything.

#60804 on its own doesn't fix much. See the second screenshot in #60978 for how pipes are still broken with #60804 merged.

Rebased, updated, and verified that (at least) everything in [BASIC PIPING GIFs](https://imgur.com/a/BDueHKa) now works again.